### DISCUSSION TITLE
Issue with losing channels

### DISCUSSION THREAD

**Discussion by eehealy** [simonsobs/sodetlib#358] [Issues and Debugging] [answered]
I am working on the rebiasing branch of sodetlib, which I locally rebased last week, and I have been encountering an issue in which channels are not responding to bias lines. 

We have detectors open to the room with NDFs. Here is a bias group map from earlier today with >300 channel unassigned, which is at least double we have been seeing in testing in recent months:
<img width="1014" alt="bias_group_map1" src="https://github.com/simonsobs/sodetlib/assets/33061247/a705620f-3193-4a72-934f-805a75207c2f">

I spent most the day operating the detectors and changing the loading (which included an LN2 load) to test out the new rebias with bias step function. After operating the detectors with the LN2 load, the noise went way up and some bias lines lost all their channels. Here's the noise after those tests:

![image-20230614-190302 (1)](https://github.com/simonsobs/sodetlib/assets/33061247/decb8262-32e7-4c9a-829d-e874def567bc)

After relocking, the noise went up but even more channels are now missing from the bias group mapping. 
![image (170)](https://github.com/simonsobs/sodetlib/assets/33061247/5e299767-f840-43d9-bf3f-5d96d7d11234)

So I think there is some issue that is causing channels not to respond to a bias line, but I don't know how to solve this problem.

**Comment by msilvafe** [comment]
I feel like we had some similar issues in P10R2 with the high loading relative to Psat 90 GHz bolos. Can you paste some of the following plots here to help debug:
1. `plot_step_fits` for a few channels that are assigned None
2. `plot_iv_res_comparison`
3. `plot_Rtes`

**Reply by eehealy** [reply]
unassigned channel plot_step_fits:
![image](https://github.com/simonsobs/sodetlib/assets/33061247/84f2b06c-6b01-4dbb-aa83-b7372534f6a1)
![image](https://github.com/simonsobs/sodetlib/assets/33061247/e2dbaafc-b71b-4863-a534-a7f50a103996)
and here's an assigned one 
![image](https://github.com/simonsobs/sodetlib/assets/33061247/802e8c35-3b74-445c-98c0-04b844c1c44e)

here's plot_iv_res_comparison
![image](https://github.com/simonsobs/sodetlib/assets/33061247/41250992-9c50-4f24-818e-c29702192500)

and here's plot_Rtes
![image](https://github.com/simonsobs/sodetlib/assets/33061247/4eaefbfa-f677-4c41-bc70-f30efd5afb0a)

**Reply by jlashner** [reply]
For the unassigned detectors, maybe can you try plotting the whole signal instead of the step fit? maybe using something like:
```
rc = np.where(bsa.bgmap == -1)[0][0]
plt.plot(bsa.am.signal[rc])
```

**Reply by jlashner** [reply]
If you could include plots from a few different unassigned channels that would be awesome

**Reply by eehealy** [reply]
here are a few
![image](https://github.com/simonsobs/sodetlib/assets/33061247/c1d6db53-eadd-4f2c-9b9f-055f62e0b459)
![image](https://github.com/simonsobs/sodetlib/assets/33061247/91cc0393-2f5d-4a61-bfdd-15c4d384ceb5)
![image](https://github.com/simonsobs/sodetlib/assets/33061247/1f42abb0-0550-4e2a-b10d-e268c8890346)
![image](https://github.com/simonsobs/sodetlib/assets/33061247/55ee3894-5374-4f99-8cf9-65a94205529d)
![image](https://github.com/simonsobs/sodetlib/assets/33061247/1240e364-20f6-4fac-bfec-9734d9b6a997)
![image](https://github.com/simonsobs/sodetlib/assets/33061247/7088fbe4-ec1f-4b39-b344-d13f80f2b1a3)
![image](https://github.com/simonsobs/sodetlib/assets/33061247/f01d09bf-280e-45a1-a433-e1df86e59ee9)

**Comment by jlashner** [comment]
Are the noise psds posted above from superconducting detectors? I think it is worth figuring out what's causing that high 1/f, that would certainly effect the bgmap determination. Looking at raw tod's and the step fits like max mentioned above might help with that.

**Reply by eehealy** [reply]
Those were taken after the detectors were exposed to a nitrogen cold load which was a big temperature swing, so I think that's probably right and the detectors might have been driven superconducting. The plots I just sent for the previous comment are after relock and taking bias steps anew.

**Comment by jlashner** [comment]
Interesting... channels that look like this are definitely a problem: 
![image](https://github.com/simonsobs/sodetlib/assets/4718487/3d75af2a-b797-4bfd-a0ee-883c7ab374f5)

To me, this looks like an issue with the tune. Usually when faced with something like this, I try to focus on a single channel and fix / understand why it looks like this. Here is what I would do to narrow down what could be causing it:

1. Choose a smurf band/channel that looks bad
2. Re-run uxm-relock. Check if the noise from that channel is still bad.
3. Look at tracking channel plots for that channel
4. Turn off flux ramp and plot the resonator response using `plot_channel_resonance`

This will give us a lot more info on what could be causing higher noise, so if you could run through that / post plots here that would be great!

**Reply by eehealy** [reply]
1. Here is a look at a channel with bad noise:
before relock:
![image](https://github.com/simonsobs/sodetlib/assets/33061247/db241873-15c1-4f6c-8f4a-f083425fc322)

2. after relock:
![image](https://github.com/simonsobs/sodetlib/assets/33061247/c7d126a7-75c5-460b-bfc4-0d602240501f)

3. What are considered the "tracking plots" -- is there a function other that plot_channel response that I should use? That one came back with no data in the plots.

4. Here's plot_channel_resonance of the same channel
![image](https://github.com/simonsobs/sodetlib/assets/33061247/4ae7c84c-2304-4fe1-8f46-c153c5a1a9fd)

**Comment by jlashner** [comment, marked as answer]
Ah ok, this is definitely the problem:
![image](https://github.com/simonsobs/sodetlib/assets/4718487/4b2619c7-95d4-4881-b943-fb89d87c7c4d)

 This resonance plot means that the phase delay SMuRF is currently using is incorrect, which induces that linear trend in the RF phase. This can sometimes happen if something changes with the RF cabling.

To fix this, I would manually re-run the following setup procedures:
1. Run `uxm_setup.setup_phase_delay`
2. Run `uxm_setup.setup_tune`
3. Plot the channel resonance again for this same channel to check that it looks better
4. Run `tracking.relock_tracking_setup`
5. Check noise

This is manually running through a subset of the operations that are run in `uxm_setup` procedure, and should hopefully get you back to decent noise.

**Reply by eehealy** [reply]
success!
![image](https://github.com/simonsobs/sodetlib/assets/33061247/1ce6ed9d-9c49-47b5-a96c-919b9c6a5df6)

but then the next command I tried lost connection to the smurf server... should i start a new discussion?

**Reply by eehealy** [reply]
RuntimeError: epics failed to respond

**Reply by jlashner** [reply]
lmao of course.... what command as it? For now if you could just post here, since I believe this has already been brought up in other discussions. Trying to re-run the command may fix it, I believe somehow the timeout on commands seems like it's been shortened and I think that's been causing the increase in failed-to-respond errors.

**Reply by eehealy** [reply]
i tried to take a bias map
bgs = np.arange(12)
bsa = bs_op.take_bgmap(S, cfg, bgs)

was there something else I was supposed to run after setting up tune?

**Reply by eehealy** [reply]
when i tried re-running it, i got:
cannot connect to smurf_server_s4:AMCc:FpgaTopLevel:AppTop:AppCore:RtmCryoDet:RtmSpiMax:TesBiasDacCtrlRegCh[1]

**Reply by jlashner** [reply]
huh did you do the tracking / noise steps? I think if you see the noise has been fixed that technically solves this discussion, and then I'd just try to re-run take_bgmap

**Reply by jlashner** [reply]
Can you do other actions like take_noise?

**Reply by eehealy** [reply]
no nothing is working, keep getting
"cannot connect to smurf_server_s4:AMCc:FpgaTopLevel:AppTop:AppCore:RtmCryoDet:RtmSpiMax:TesBiasDacCtrlRegCh[1]"

**Comment by eehealy** [comment]
EH and JL discussion over zoom
- issue that caused the crash might have been running bs without tracking! soft hammering seemed to solve the problem
- determined that the origin of this problem may have been swapping the device config file at the start of this testing. to use the re-biasing with bias steps, I had to add a line of code. The rest of the configuration file should have been identical, but perhaps it was not.

**Comment by eehealy** [comment]
ok so things look much better! i am, however, seeing electrical pickup in the noise that i wasn't seeing before:
![image](https://github.com/simonsobs/sodetlib/assets/33061247/2cd09172-072f-4ce2-ad50-4ef0e5adda38)

**Reply by msilvafe** [reply]
Are you talking about the small line at 60 Hz? That is often an indication of a shorted FR or BL.

**Reply by eehealy** [reply]
yeah that's what i was referring to. i can probe to see if we have a new short, but it wasnt there a few hours ago and nothing has been touched except through smurf.. so i dont know what could have caused a new short to appear...

**Comment by jlashner** [comment]
Great!! I think since the topic of the initial discussion has been solved, I'm going to close this, but feel free to open new discussions about the short / other topics!
